### PR TITLE
#12 Collections and Generics

### PR DESCRIPTION
**Pull request by Preethxsuresh**
Explored the use of collections and generics in C#. 
Here's the assignment document for reference.
[Understanding and Practicing Collections and Generics in C.pdf](https://github.com/Preethxsuresh/C_Sharp/files/12711967/Understanding.and.Practicing.Collections.and.Generics.in.C.pdf)

### CODE REVIEW

File: src/assignmenttwelve/ListManager.cs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+            }
+        }
+
+        /// <summary>
+        /// Removes an book from the list based on its index.
+        /// </summary>
+        /// <typeparam name="T">The type of items in the list.</typeparam>
+        /// <param name="bookList">The list from which to remove an book.</param>
+        /// <returns>The updated list with the book removed, or the original list if removal fails.</returns>
+        public static List<T> RemoveItemAtIndex(List<T> bookList)
```
Comment (nandhakumarsoliton) [inline src/assignmenttwelve/ListManager.cs:88]: Index should also be passed as a parameter to the function `RemoveItemAtIndex`.
Else we can have a suitable name for this function.

File: src/assignmenttwelve/ListManager.cs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        /// <summary>
+        /// Prompts user to add, remove, search and display list items.
+        /// </summary>
+        /// <returns><c>True</c> when list operations are performed; otherwise <c>False</c>.</returns>
+        public static bool PerformListOperations()
+        {
+            // Creates a new list.
+            List<T> bookList = new List<T>();
+
+            AddItemsToList(bookList, 5);
```
Comment (nandhakumarsoliton) [inline src/assignmenttwelve/ListManager.cs:148]: `AddItemsToList` returns a bookList. Are we using it anywhere?

File: src/assignmenttwelve/QueueManager.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        /// </summary>
+        /// <typeparam name="T">The type of elements in the queue.</typeparam>
+        /// <param name="queue">The queue from which elements will be removed.</param>
+        /// <returns>The modified queue with elements removed.</returns>
+        public static Queue<T> RemovePersonsFromQueue(Queue<T> queue)
+        {
+            Utility.SetAlertTextColor("\nHere's the Queue,", "Green");
+            while (queue.Count > 0)
+            {
+                DisplayQueue(queue);
```
Comment (nandhakumarsoliton) [inline src/assignmenttwelve/QueueManager.cs:44]: Why are we displaying the queue inside Remove function?

File: src/assignmenttwelve/Utility.cs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            return default(T);
+        }
+
+        /// <summary>
+        /// Modifies the text color.
+        /// </summary>
+        /// <param name="alertText">string Message.</param>
+        /// <param name="alertTextColor">color to modify.</param>
+        /// <returns>Always return true.</returns>
+        public static bool SetAlertTextColor(string alertText, string alertTextColor)
```
Comment (nandhakumarsoliton) [inline src/assignmenttwelve/Utility.cs:78]: Can we have a better function name? `SetAlertTextColor` means it as a setter function which will not return anything and sets certain value to a property/variable.
Here we are Displaying some message to user in certain colour.

File: src/assignmenttwelve/Utility.cs [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            Console.ForegroundColor = ConsoleColor.White;
+            return true;
+        }
+
+        /// <summary>
+        /// Calculates the sum of given elements in the collection.
+        /// </summary>
+        /// <param name="collection">Set of elements in list, queue or array.</param>
+        /// <returns>Sum of elements.</returns>
+        public static int SumOfElements(IEnumerable<int> collection)
```
Comment (nandhakumarsoliton) [inline src/assignmenttwelve/Utility.cs:92]: This can be a Getter function -> `GetSumOfElements`.

File: src/assignmenttwelve/QueueManager.cs (hunk 3 above)
Comment (Preethxsuresh) [reply src/assignmenttwelve/QueueManager.cs:44]: It's just to display that the queue elements are removed one by one until it is empty

**Review by nandhakumarsoliton — APPROVED** [review]

**Pull request merged**